src/visitors/cvode_visitor.cpp

```diff
@@ -0,0 +1,200 @@
+        in_differential_equation = false;
+    }
+};
+
+class NonStiffVisitor: public CvodeHelperVisitor {
+  public:
+    explicit NonStiffVisitor(symtab::SymbolTable* symtab) {
+        program_symtab = symtab;
+    }
+
+    inline void visit_binary_expression(ast::BinaryExpression& node) {
+        const auto& lhs = node.get_lhs();
+
+        if (!in_differential_equation || !lhs->is_var_name()) {
+            return;
+        }
+
+        auto name = std::dynamic_pointer_cast<ast::VarName>(lhs)->get_name();
+        auto varname = cvode_set_lhs(node);
+
+        if (program_symtab->lookup(varname) == nullptr) {
+            auto symbol = std::make_shared<symtab::Symbol>(varname, ModToken());
+            symbol->set_original_name(name->get_node_name());
+            program_symtab->insert(symbol);
+        }
+    }
+};
+
+class StiffVisitor: public CvodeHelperVisitor {
+  public:
+    explicit StiffVisitor(symtab::SymbolTable* symtab) {
+        program_symtab = symtab;
+    }
+
+    inline void visit_binary_expression(ast::BinaryExpression& node) {
+        const auto& lhs = node.get_lhs();
+
+        if (!in_differential_equation || !lhs->is_var_name()) {
+            return;
+        }
+
+        auto name = std::dynamic_pointer_cast<ast::VarName>(lhs)->get_name();
+        auto varname = cvode_set_lhs(node);
+
+        if (program_symtab->lookup(varname) == nullptr) {
+            auto symbol = std::make_shared<symtab::Symbol>(varname, ModToken());
+            symbol->set_original_name(name->get_node_name());
+            program_symtab->insert(symbol);
+        }
+
+        auto rhs = node.get_rhs();
+        // map of all indexed symbols (need special treatment in SymPy)
+        auto name_map = get_name_map(*rhs, name->get_node_name());
+        auto diff2c = pywrap::EmbeddedPythonLoader::get_instance().api().diff2c;
+        auto [jacobian,
+              exception_message] = diff2c(to_nmodl(*rhs), parse_independent_var(name), name_map);
+        if (!exception_message.empty()) {
+            logger->warn("CvodeVisitor :: python exception: {}", exception_message);
+        }
+        // NOTE: LHS can be anything here, the equality is to keep `create_statement` from
+        // complaining, we discard the LHS later
+        auto statement = fmt::format("{} = {} / (1 - dt * ({}))", varname, varname, jacobian);
+        logger->debug("CvodeVisitor :: replacing statement {} with {}", to_nmodl(node), statement);
+        auto expr_statement = std::dynamic_pointer_cast<ast::ExpressionStatement>(
+            create_statement(statement));
+        const auto bin_expr = std::dynamic_pointer_cast<const ast::BinaryExpression>(
+            expr_statement->get_expression());
+        node.set_rhs(std::shared_ptr<ast::Expression>(bin_expr->get_rhs()->clone()));
+    }
+};
+
+
+void CvodeVisitor::visit_program(ast::Program& node) {
+    auto der_blocks = collect_nodes(node, {ast::AstNodeType::DERIVATIVE_BLOCK});
+    if (!der_blocks.empty()) {
```

Comment:
There's three cases here:
1. No derivative blocks: just return.
2. More than one block: crash loudly.
3. Continue to create the cvode block.

Reply:
This also fails for the steady state usecase, since one of the intermediate mod files has:

```plaintext
DERIVATIVE state {
    LOCAL kf0_, kb0_
    kf0_ = kf
    kb0_ = kb
    CONSERVE Z = (8.0*volB+1.0*volC-(volB*1*Y))/(volC*1)
    X' = ((-1*(kf0_*X*Y-kb0_*Z)))/(volA)
    Y' = ((-1*(kf0_*X*Y-kb0_*Z)))/(volB)
    Z' = ((1*(kf0_*X*Y-kb0_*Z)))/(volC)
}

DERIVATIVE state_steadystate {
    dt = 1000000000
    LOCAL kf0_, kb0_
    kf0_ = kf
    kb0_ = kb
    CONSERVE Z = (8.0*volB+1.0*volC-(volB*1*Y))/(volC*1)
    X' = ((-1*(kf0_*X*Y-kb0_*Z)))/(volA)
    Y' = ((-1*(kf0_*X*Y-kb0_*Z)))/(volB)
    Z' = ((1*(kf0_*X*Y-kb0_*Z)))/(volC)
}
```

I could add a `_steadystate` detector, but is it really necessary?

Reply:
It feels like an easy improvement. Now you know that you're making the assumption right here. In the future a different dev might be tasked with allowing multiple blocks, they'll remove the first check; and then need to search for hours why there's somethings just not showing up. If you add the check it'll immediately pop out to that future dev. It also serves as a reminder that this doesn't work (which is perfectly fine) and isn't intended to work for multiple derivative block, therefore the reader can put themselves into the same mindset.

If a dev dares add a derivative block, this part might start to fail spuriously, depending on in which order the blocks are 'collect'ed.

Reply:
Done.